### PR TITLE
API-40281-sp-issues-pact-previous-rated-disability

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds validation rule that a special issues should not be labeled as a PACT if the action type is 'increase'.
- Adds test.

## Related issue(s)

- [API-40281](https://jira.devops.va.gov/browse/API-40281)

## Testing done

- [X] *New code is covered by unit tests*
- Testing notes for Postman
1. Send a disability with the following:
```
                {
                    "name": "Cancer - Musculoskeletal - Knee",
                    "exposureOrEventOrInjury": "EXPOSURE",
                    "approximateDate": "2015",
                    "serviceRelevance": "ABCDEFG",
                    "disabilityActionType": "INCREASE",
                    "classificationCode": "9014",
                    "ratedDisabilityId": "ABCDEFGHIJKLMNOPQRSTUVWX",
                    "diagnosticCode": 249470,
                    "isRelatedToToxicExposure": true
                }
```
2. place a debugger at the end of the evss mapper to check if the special issues includes 'PACT' for the disability that has an action type as 'increase'

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
	modified:   modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb
	modified:   modules/claims_api/spec/lib/claims_api/v2/disability_compensation_evss_mapper_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/spec/lib/claims_api/v2/disability_compensation_evss_mapper_spec.rb [hunk 1]
```diff
@@ -201,6 +201,36 @@
...
+            isRelatedToToxicExposure: true,
+            exposureOrEventOrInjury: '',
+            ratedDisabilityId: '',
+            diagnosticCode: 0,
+            secondaryDisabilities: nil,
+            specialIssues: %w[POW EMP]
+          }
+        end
+
+        it 'maps the special issues attributes correctly and does NOT appends PACT' do
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/lib/claims_api/v2/disability_compensation_evss_mapper_spec.rb:221]: Not: `append` not `appends`

File: modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb [hunk 2]
```diff
@@ -105,7 +105,8 @@ def transform_disability_values!(disability)
 
       def check_for_pact_special_issue(disability)
         related_to_toxic_exposure = disability[:isRelatedToToxicExposure]
-        if related_to_toxic_exposure
+        action_not_increase = disability[:disabilityActionType] != 'INCREASE'
+        if related_to_toxic_exposure && action_not_increase
```
Comment (rockwellwindsor-va) [inline modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb:109]: @stiehlrod  Wonder if this check might be better up one level.  Not sure this method really needs to know about this?
Something like
```
        check_for_pact_special_issue(disability) if disability[:disabilityActionType] != 'INCREASE'
```
on line 101?

thoughts?

Comment (stiehlrod) [reply modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb:109]: @rockwellwindsor-va Good thought. thanks

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**